### ISSUE TITLE
Segmentation Fault in `torch._C._nn.thnn_conv2d` Due to Invalid Padding Argument

### ISSUE THREAD

**Issue by Gwihwan-Go** [pytorch/pytorch#121188]
### 🐛 Describe the bug

A segmentation fault occurs when invoking `torch._C._nn.thnn_conv2d` with an empty list for the padding argument, alongside other parameters that are potentially mismatched or improperly specified. 

## Error log

```bash
Segmentation fault (core dumped)
```

## Minimized Repro

```python
import torch
torch._C._nn.thnn_conv2d(torch.rand([1]), bias=torch.rand([3]), kernel_size=[5], padding=[], stride=[9], weight=torch.rand([9]))
```

### Versions

PyTorch version: 2.3.0.dev20240301+cu118
Is debug build: False
CUDA used to build PyTorch: 11.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.3 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.35

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki @malfet

**Comment by albanD** [comment]
Thanks for the report. We would definitely accept a PR adding the appropriate error checking to avoid the segfault!

**Comment by Martim03** [comment]
Hi, I managed to replicate the issue and I would be willing to solve it if you don't mind! :)

**Comment by albanD** [comment]
Sounds good. Feel free to send a PR and link this issue as fixed in the PR description

**Comment by Martim03** [comment]
Hi again, upon reviewing the code, I've noticed that a Segmentation Fault could also occur if the function is invoked with an invalid kernel_size or stride. Would you like me to include error checking for those cases as well?
